features/270-next-skycli/api.md

```diff
@@ -0,0 +1,139 @@
+
+- skycli user (Admin only, cluster user management)
+    - [skycli user create](commands.md#skycli-user-create)
+    - [skycli user list](commands.md#skycli-user-list)
+
+#### Application commands
+
+- skycli app
+    - [skycli app list](commands.md#skycli-app-list)
+    - [skycli app create](commands.md#skycli-app-create)
+    - [skycli app add-user [USER_EMAIL]](commands.md#skycli-app-add-user)
+    - [skycli app view-tenant-config](commands.md#skycli-app-view-config)
+    - [skycli app update-tenant-config -f [TENANT_CONFIG_YAML_FILE]](commands.md#skycli-app-update-config)
+
+- skycli domain
+    - [skycli domain add [CUSTOM_DOMAIN]](commands.md#skycli-domain-create)
+    - [skycli domain verify [CUSTOM_DOMAIN]](commands.md#skycli-domain-verify)
+    - [skycli domain update [CUSTOM_DOMAIN]](commands.md#skycli-domain-update)
+    - [skycli domain set-alias [CUSTOM_DOMAIN] [CF_VERSIONED_LINK]](commands.md#skycli-domain-set-alias)
+    - [skycli domain list](commands.md#skycli-domain-list)
+
+- skycli secret
+    - [skycli secret list](commands.md#skycli-secret-list)
+    - [skycli secret create [SECRET_NAME] [SECRET_VALUE]](commands.md#skycli-secret-create)
+    - [skycli secret update [SECRET_NAME] [SECRET_VALUE]](commands.md#skycli-secret-update)
+    - [skycli secret remove [SECRET_NAME]](commands.md#skycli-secret-remove)
+
+- skycli service
+    - [skycli service list](commands.md#skycli-service-list)
+    - [skycli service enable [SERVICE_NAME] [SERVICE_PARAMS...]](commands.md#skycli-service-enable)
+
+## Commands alias
+
+- skycli login -> skycli auth login
+- skycli logout -> skycli auth logout
+
+## Architecture
+
+To keep it simple, all clusters user have same privilege. They can create app
+and update their app settings or cf deployment. Cluster admin use admin key to
+access skycli instead of using user account.
+
+```
+                         +----------+
+                         |          |
+                         |  Skycli  |
+                         |          |
+                         +----+-----+
+                              |
+                              |
+                         +----v-----------------+
+                         |                      |
+                         |  Cluster controller  |
+                         |                      |
+                         +-+------------------+-+
+- Cluster user store       |                  |
+- Update tenant config     |                  |  Update deployment
+- Store build history?     |                  |
+                         +-v------------+    +v----------+
+                         |              |    |           |
+                         | gateway db   |    |    k8s    |
+                         |              |    |           |
+                         +--------------+    +-----------+
+
+```
+
+## Config files structure
+
+### ~/.skycli/skyclirc
+
+- cluster controller endpoint
+- user credentials 
+
+### ./.skyclirc
```

Comment:
will it try to go to the root git repository to find the .skyclirc? Or it is a bad practices? I don't know...